ow_power_system/src/PowerSystemNode.cpp

```diff
@@ -751,15 +756,119 @@ void PowerSystemNode::injectFault (const std::string& fault_name)
       }
     }
   }
+
+  // Low state of charge case
+  if (fault_name == FAULT_NAME_LOW_SOC)
+  {
+    // Check if the fault has switched.
+    if (!m_low_soc_activated && fault_enabled)
+    {
+      // Block the fault if the power system is disabled.
+      if (!m_enable_power_system)
+      {
+        printFaultDisabledWarning();
+        return;
+      }
+
+
+      ROS_INFO_STREAM(fault_name << " activated!");
+      m_low_soc_activated = true;
+      // Warn the user about the invalid behavior associated with power faults.
+      ROS_WARN_STREAM("Note that artificial power fault injection simply "
+                   << "temporarily overrides GSAP's predictions. RUL will "
+                   << "not react to the fault and the system "
+                   << "will continue without any loss of battery health after "
+                   << "the fault is removed, which is not realistic.");
```

Comment:
This repeated message should be factored into one place, and I suggest this rewording: 

"Note that artificial power fault injection simply overrides GSAP's predictions. Remaining Useful Life will not be affected and the system will continue without any loss of battery health after the fault is cleared."

Also, you can break this up into short lines simply by juxtaposing strings, e.g. 
```
"foo" 
" bar" 
```
instead of separating them with `<<`, but this is a minor point.